pkg/api/routes.go

```diff
@@ -58,14 +58,14 @@ func NewRouteHandler(c *Controller) *RouteHandler {
 }
 
 func (rh *RouteHandler) SetupRoutes() {
-	rh.c.Router.Use(AuthHandler(rh.c))
+	prefixedRouter := rh.c.Router.PathPrefix(RoutePrefix).Subrouter()
```

Comment:
with this change I think we will be authorizing routes starting from /v2 only and /query will be skipped.

Reply:
yes, currently a draft PR, still evaluating implications.

Reply:
This is fixed now. Pls take a look.